src/com/maddyhome/idea/vim/extension/commentary/CommentaryExtension.java

```diff
@@ -0,0 +1,138 @@
+ */
+public class CommentaryExtension extends VimNonDisposableExtension {
+
+  @NotNull
+  @Override
+  public String getName() {
+    return "commentary";
+  }
+
+  @Override
+  protected void initOnce() {
+    putExtensionHandlerMapping(MappingMode.N, parseKeys("<Plug>(CommentMotion)"), new CommentMotionHandler(), false);
+    putExtensionHandlerMapping(MappingMode.N, parseKeys("<Plug>(CommentLine)"), new CommentLineHandler(), false);
+    putExtensionHandlerMapping(MappingMode.VO, parseKeys("<Plug>(CommentMotionV)"), new CommentMotionVHandler(), false);
+
+    putKeyMapping(MappingMode.N, parseKeys("gc"), parseKeys("<Plug>(CommentMotion)"), true);
+    putKeyMapping(MappingMode.N, parseKeys("gcc"), parseKeys("<Plug>(CommentLine)"), true);
+    putKeyMapping(MappingMode.VO, parseKeys("gc"), parseKeys("<Plug>(CommentMotionV)"), true);
+  }
+
+  private static class CommentMotionHandler implements VimExtensionHandler {
+    @Override
+    public void execute(@NotNull Editor editor, @NotNull DataContext context) {
+      setOperatorFunction(new Operator());
+      executeNormal(parseKeys("g@"), editor);
+    }
+  }
+
+  private static class CommentMotionVHandler implements VimExtensionHandler {
+    @Override
+    public void execute(@NotNull Editor editor, @NotNull DataContext context) {
+      final TextRange visualRange = VimPlugin.getMark().getVisualSelectionMarks(editor);
+      if (visualRange == null) {
+        return;
+      }
+
+      // always use line-wise comments
+      if (!new Operator().apply(editor, context, SelectionType.LINE_WISE)) {
+        return;
+      }
+
+      // Leave visual mode
+      executeNormal(parseKeys("<Esc>"), editor);
+
+      editor.getCaretModel().moveToOffset(visualRange.getStartOffset());
+    }
+  }
+
+  private static class Operator implements OperatorFunction {
+    @Override
+    public boolean apply(@NotNull Editor editor, @NotNull DataContext context, @NotNull SelectionType selectionType) {
+      final TextRange range = getCommentRange(editor);
+      if (range == null) return false;
+
+      if (CommandState.getInstance(editor).getMode() != CommandState.Mode.VISUAL) {
+        editor.getSelectionModel().setSelection(range.getStartOffset(), range.getEndOffset());
+      }
+
+      final MultiCaretCodeInsightActionHandler handler =
+        selectionType == SelectionType.CHARACTER_WISE
+          ? new CommentByBlockCommentHandler()
+          : new CommentByLineCommentHandler();
+
+      try {
+        Project proj = editor.getProject();
+        if (proj == null) return false;
+
+        PsiFile file = PsiDocumentManager.getInstance(proj).getPsiFile(editor.getDocument());
+        if (file == null) return false;
+
+        handler.invoke(editor.getProject(), editor, editor.getCaretModel().getCurrentCaret(), file);
+        handler.postInvoke();
+
+        // Jump back to start
+        executeNormal(parseKeys("`["), editor);
+        return true;
+      } catch (RuntimeException e) {
```

Comment:
Please remove this catch. Occurring runtime exceptions are captured by IntelliJ IDEA and can be sent to JetBrains. But it is not possible if exception is suppressed.